### ISSUE TITLE
PropertyBuilder: Should there be a "break" here?

### ISSUE THREAD

**Issue by douglm** [ical4j/ical4j#723]
**Describe the bug**
In PropertyBuilder (4.0.3) we have this code:

`
        for (PropertyFactory<?> factory : factories) {
            if (factory.supports(name)) {
                property = factory.createProperty(new ParameterList(parameters), value);
                if (property instanceof DateProperty) {
                    DateProperty<?> dateProp = (DateProperty<?>) property;
                    // don't set timezone on UTC-formatted properties..
                    if (!(dateProp instanceof UtcProperty)) {
                        dateProp.setTimeZoneRegistry(timeZoneRegistry);
                        dateProp.setDefaultTimeZone(defaultTimeZone);
                        property.setValue(value);
                    }
                } else if (property instanceof DateListProperty) {
                    DateListProperty<?> dateListProperty = (DateListProperty<?>) property;
                    dateListProperty.setTimeZoneRegistry(timeZoneRegistry);
                    dateListProperty.setDefaultTimeZone(defaultTimeZone);
                    property.setValue(value);
                }
            }
        }
`
Shouldn't there be a break inside that "if (factory.supports(name)) {...}"
Seems we're just iterating over all the factories every time.

Is there any situation where we want more than one factory?

If not - couldn't we optimize this with a map of property-name to factory?

**To Reproduce**

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Windows, Linux, Android]
 - Java Version [e.g. 7, 8, 9]
 - iCal4j Version

**Additional context**

**Comment by benfortuna** [comment]
Yeah, I think the main reason for the seemingly convoluted approach was to support model extensions found on the classpath. 

The idea was to allow extension factories to support additional properties, and I suppose to replace default factories. Without having a `break` in the loop should end up with the last found factory providing the implementation, but in hindsight that may not be guaranteed.

So by adding a `break` it would stop at the first found factory, which I think is fine.

I can also see why a `Map` would make sense for the finite set of known properties, but perhaps doesn't support the idea of extension property factories on the classpath..
